### PR TITLE
Fix Bark logits processors device misplacement

### PR DESCRIPTION
**Pull request by ylacombe**
# What does this PR do?
Fix some Bark failing tests where the logits processors' tensors were misplaced on CPU!
It was introduced by #29804 but not flagged because of the Bark custom generation!

Slow tests are now all green on a A100!

cc @ydshieh and @amyeroberts

### CODE REVIEW

**Review by amyeroberts — COMMENTED** [review]
Thanks for working on a fix! 

~I'm a bit confused by this solution - neither [SuppressTokensLogitsProcessor](https://github.com/ylacombe/transformers/blob/daf281f44f654abd5e7ed07e37985cee2250c3af/src/transformers/generation/logits_process.py#L1821) nor [BarkEosPrioritizerLogitsProcessor](https://github.com/ylacombe/transformers/blob/daf281f44f654abd5e7ed07e37985cee2250c3af/src/transformers/generation/logits_process.py#L2295) take `device` as an init kwarg~ My bad - I was looking at an older commit

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing! 

Tbh, this pattern of passing the device here I think is a bit odd ([why do we take a tensor, turn it into a list, then turn it back into a tensor on the device](https://github.com/ylacombe/transformers/blame/b7672826cad31e30319487af876e608d8af7d37b/src/transformers/generation/logits_process.py#L1784)? but that's more of a generate question)

**Comment by amyeroberts** [comment]
> It was introduced by https://github.com/huggingface/transformers/pull/29804 but not flagged because of the Bark custom generation!

@ylacombe Does this mean that Bark's generation isn't properly tested?

**Comment by ylacombe** [comment]
> @ylacombe Does this mean that Bark's generation isn't properly tested?

It's not at the moment, we've discussed the reasons why in [this thread](https://github.com/huggingface/transformers/pull/25372#discussion_r1287026171). It might make sense to revisit this decision once the standard generation method is refactored, WDYT ?

**Comment by ydshieh** [comment]
Thank you @ylacombe !

@amyeroberts  There is still `BarkModelIntegrationTests` and it catches the bug :-) which leads to this PR.

**Review by ydshieh — APPROVED** [review]
💯
